--- hdf-eos2-3.0/src/GDapi.c.orig	2021-05-18 13:55:16.000000000 +0200
+++ hdf-eos2-3.0/src/GDapi.c	2024-12-21 21:37:45.646176303 +0100
@@ -87,6 +87,7 @@ Dec  10  2013   Abe Taaheri  Increased N
                              respectively.
 Jul  09, 2020   Peter Smith  Updated szlib code exclusions
 ******************************************************************************/
+#include "HE2_config.h"
 #include "stdio.h"
 #include "mfhdf.h"
 #include "hcomp.h"
--- hdf-eos2-3.0/src/SWapi.c.orig	2024-12-21 21:38:01.889421639 +0100
+++ hdf-eos2-3.0/src/SWapi.c	2024-12-21 21:35:57.930093184 +0100
@@ -74,6 +74,7 @@ Jul  09, 2020   Peter Smith  Updated szl
 ******************************************************************************/
 #include <stdio.h>
 #include <stdlib.h>
+#include "HE2_config.h"
 #include "mfhdf.h"
 #include "hcomp.h"
 #include "HdfEosDef.h"
